File: ABC-242/C.py
import sys

sys.setrecursionlimit(10**8)

MOD = 998244353

n = int(input())

cnt = 0


def dfs(A: str):
    if len(A) == n:
        global cnt
        cnt += 1
        cnt %= MOD
        return

    for di in [-1, 0, 1]:
        if int(A[-1]) == 1 and di == -1:
            continue
        if int(A[-1]) == 9 and di == 1:
            continue

        A += str(int(A[-1]) + di)
        dfs(A)
        A = A[:-1]


for i in range(1, 10):
    dfs(str(i))
print(cnt)

File: ABC-242/C.cpp
#include <iostream>
#include <string>

using namespace std;
int cnt, n;
int MOD = 998244353;

void dfs(string A) {

  if (A.length() == n) {
    cnt++;
    cnt %= MOD;
    return;
  }

  for (int di = -1; di < 2; di++) {
    int _n = A.length();
    if (A[_n - 1] == '1' && di == -1) {
      continue;
    }
    if (A[_n - 1] == '9' && di == 1) {
      continue;
    }

    A += to_string(int(A[_n - 1] - '0') + di);
    cout<<A<<endl;
    dfs(A);
    A = A.substr(0, _n - 1);
  }
}

int main() {
  cin >> n;
  for (int i = 1; i < 10; i++) {
    dfs(to_string(i));
  }
  cout << cnt << endl;
  return 0;
}
